### PR TITLE
2964 add notification for devfile registry update

### PR DESCRIPTION
**Pull request by msivasubramaniaan**
Fixes: https://github.com/redhat-developer/vscode-openshift-tools/issues/2964

The PR does below two things:

1. 
     registry-notification
2.
    registry-not-valid-url

### CODE REVIEW

File: src/registriesView.ts [hunk 1]
```diff
@@ -320,10 +321,22 @@ export class ComponentTypesView implements TreeDataProvider<ComponentType> {
...
+            const componentTypes = JSON.parse(await response.text()) as DevfileComponentType[];
+            if (componentTypes.length > 0) {
+                void Progress.execFunctionWithProgress('Devfile registry is updating',async () => {
+                    const newRegistry = await OdoImpl.Instance.addRegistry(regName, regURL, token);
+                    ComponentTypesView.instance.addRegistry(newRegistry);
+                    await ComponentTypesView.instance.getAllComponents();
+                    ComponentTypesView.instance.refresh(false);
+                })
+            }
+        } catch (reson) {
```
Comment (mohitsuman) [inline src/registriesView.ts:337]: What does reson mean here ?

**Review by mohitsuman — CHANGES_REQUESTED** [review]
fix error handling

File: src/registriesView.ts [hunk 2]
```diff
@@ -320,10 +321,22 @@ export class ComponentTypesView implements TreeDataProvider<ComponentType> {
...
+            void vscode.window.showErrorMessage(`Invalid registry URL ${regURL}`);
```
[the remark this answers is not in the document]
Comment (mohitsuman) [reply src/registriesView.ts:338]: We also should add a check to see if the registry URL is duplicate or not and notify the user accordingly.

[the remark this answers is not in the document]
Comment (msivasubramaniaan) [reply src/registriesView.ts:338]: Already handled that scenario by https://github.com/redhat-developer/vscode-openshift-tools/pull/2966

**Pull request merged**